mdn/developer-portal
# 1604: Marker for mozilla sponsored events

This changeset adds a way to mark certain events as 'Sponsored by Mozilla', and shows that in the Events list and on the Event detail

UPDATED: to support a custom label, with a fallback phrase if no one is provided.

(Related issue #1604)

## Screenshots
![Screenshot 2020-07-01 at 14 11 51](https://user-images.githubusercontent.com/101457/86260674-81d64780-bbb5-11ea-9884-6fd47df98d93.png)

![Screenshot 2020-07-01 at 14 17 32](https://user-images.githubusercontent.com/101457/86260988-e8f3fc00-bbb5-11ea-8126-241038bd75fc.png)

![Screenshot 2020-07-01 at 14 17 40](https://user-images.githubusercontent.com/101457/86260992-ea252900-bbb5-11ea-8672-235da18aff10.png)

![Screenshot 2020-07-01 at 14 17 44](https://user-images.githubusercontent.com/101457/86260993-ea252900-bbb5-11ea-9882-bcc5ce20c692.png)

![Screenshot 2020-07-01 at 14 29 24](https://user-images.githubusercontent.com/101457/86260996-ea252900-bbb5-11ea-93e6-6106d374a7c3.png)

![Screenshot 2020-07-01 at 15 49 12](https://user-images.githubusercontent.com/101457/86260999-eabdbf80-bbb5-11ea-8649-1c4079d22838.png)

![Screenshot 2020-07-01 at 15 49 24](https://user-images.githubusercontent.com/101457/86261003-eb565600-bbb5-11ea-8f6c-f7df6c40db95.png)






## Key changes:

* Adds a BooleanField to mark if an Event was backed by Mozilla
* Adds a CharField (optional) to allow describing how the event was backed by Mozilla
* Adds a label atom for showing this message
* Use label atom on the Events listing page, in the relevant event-card components
* Use the label on the Event detail page itself by pre-rendering the relevant HTML and passing it into the header strip as a marked-safe HTML string

## How to test

- Code is deployed to stage. 
- Mark some Events as sponsored and confirm that the events listing shows it, as well as the event detail
- Try out the custom label option, too
- Confirm that non-sponsored events don't show the sponsored label in the detail view or the events listing
